pkg/liqoctl/status/remoteinfo.go

```diff
@@ -0,0 +1,176 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	controllerRuntimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/liqotech/liqo/pkg/utils/getters"
+)
+
+// RemoteInfoChecker implements the Check interface.
+// holds informations about remote clusters.
+type RemoteInfoChecker struct {
+	client             controllerRuntimeClient.Client
+	namespace          string
+	clusterNameFilter  *[]string
+	clusterIDFilter    *[]string
+	errors             bool
+	rootRemoteInfoNode InfoNode
+	collectionErrors   []collectionError
+}
+
+// newRemoteInfoChecker return a new remote info checker.
+func newRemoteInfoChecker(namespace string, clusterNameFilter, clusterIDFilter *[]string, client controllerRuntimeClient.Client) *RemoteInfoChecker {
+	return &RemoteInfoChecker{
+		client:             client,
+		namespace:          namespace,
+		clusterNameFilter:  clusterNameFilter,
+		clusterIDFilter:    clusterIDFilter,
+		errors:             false,
+		rootRemoteInfoNode: newRootInfoNode("Remote Clusters Information"),
+	}
+}
+
+// argsFilterCheck returns true if the given foreignClusterID or foreignClusterName are contained
+// in the respective filter list or if filter lists are void.
+func (ric *RemoteInfoChecker) argsFilterCheck(foreignClusterID, foreignClusterName string) bool {
+	return stringArrayContainsString(ric.clusterIDFilter, foreignClusterID) ||
+		stringArrayContainsString(ric.clusterNameFilter, foreignClusterName) ||
+		(len(*ric.clusterIDFilter) == 0 && len(*ric.clusterNameFilter) == 0)
+}
+
+// stringArrayContainsString returns true if the given string is contained in the given array.
+func stringArrayContainsString(a *[]string, s string) bool {
+	for _, v := range *a {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+// Collect implements the collect method of the Checker interface.
+// it collects the infos of the Remote cluster.
+func (ric *RemoteInfoChecker) Collect(ctx context.Context) error {
+	var localNetworkConfigNode, remoteNetworkConfigNode, selectedNode *InfoNode
+	var remoteNodeMsg string
+	localClusterIdentity, err := getLocalClusterIdentity(ctx, ric.client, ric.namespace)
+	localClusterName := localClusterIdentity.ClusterName
+	if err != nil {
+		ric.addCollectionError("LocalClusterName", "", err)
+		ric.errors = true
+	}
+
+	networkConfigs, err := getters.GetNetworkConfigsByLabel(ctx, ric.client, "", labels.NewSelector())
+	if err != nil {
+		ric.addCollectionError("NetworkConfigs", "unable to collect NetworkConfigs", err)
+		ric.errors = true
+	}
```

Comment:
A simpler logic would be to filter the foreign clusters in advance, and then for each one search for the desired resources in the specific tenant namespace (stated as part of the status).